File: framework/languages/dotnet/e2e/RegistrationCodec/Server/CodecRequester/CodecRequesterOptions.cs
namespace RegistrationCodec.Server.CodecRequester;

using Zlink.Framework.E2E.Configuration;

internal sealed record CodecRequesterOptions(
    string Rid,
    string HttpUrl,
    string ChannelEndpoint,
    string LogDir)
{
    public static CodecRequesterOptions Parse(string[] args)
        => E2eConfiguration.Load<CodecRequesterOptions>(args);
}

File: framework/languages/java/e2e/RegistrationCodec/Server/CodecRequester/src/main/java/systems/zlink/e2e/registrationcodec/codecrequester/Configuration/CodecRequesterOptions.java
package systems.zlink.e2e.registrationcodec.codecrequester.Configuration;

public record CodecRequesterOptions(
    String serverEndpoint,
    String httpEndpoint,
    String logDir) {
    public static CodecRequesterOptions fromEnv() {
        return new CodecRequesterOptions(
            Env.get("ZLINK_JAVA_E2E_SERVER_ENDPOINT"),
            Env.get("ZLINK_JAVA_E2E_HTTP_ENDPOINT"),
            Env.get("ZLINK_JAVA_E2E_LOG_DIR", "logs"));
    }
}
